pkg/cosign/verify.go

```diff
@@ -343,3 +350,35 @@ func correctAnnotations(wanted, have map[string]string) bool {
 	}
 	return true
 }
+
+func verifyBundle(bundle Bundle) (bool, error) {
+
+	// TODO: Verify the signed inclusion proof against rekor public key
+
+	// TODO: verify rekor intermediate cert came from fulcio via https://github.com/google/certificate-transparency-go/blob/109466f2efc1e5eeb78c9bc25cbc6069b8afa40b/trillian/ctfe/cert_checker.go#L49
+
+	if bundle.InclusionProof == "" {
+		return false, nil
+	}
+
+	ip := models.InclusionProof{}
+	if err := ip.UnmarshalBinary([]byte(bundle.InclusionProof)); err != nil {
+		return false, errors.Wrap(err, "unmarshalling binary")
+	}
+
+	hashes := [][]byte{}
+	for _, h := range ip.Hashes {
+		hb, _ := hex.DecodeString(h)
+		hashes = append(hashes, hb)
+	}
+
+	rootHash, _ := hex.DecodeString(*ip.RootHash)
+	leafHash, _ := hex.DecodeString(bundle.UUID)
```

Comment:
Couldn't I use an existing bundle from any other image and it would validate just fine? This assumes the image uploader wants to pretend they logged the image but actually hasn't.

Reply:
@priyawadhwa is on vacation this week but I think this entire change is getting rearchitected somewhere else. I'll wait for her to get back and confirm to close it though.

Thanks for taking a look though!

Cc'ing @asraa in case she has a link to the new idea somewhere.

Reply:
Yup, this PR is a bit out of date now. We've decided to add the following to rekor first:
- Return a signed timestamped "promise" that an entry was included in the log, similar to how a CT log returns an SCT

This promise will be included in the bundle. Then, cosign should be able to validate this promise offline. 

I'm going to go ahead and close this PR for now, and then probably open a new one once the new stuff in rekor has been merged!